sig-scheduler-plugins/apis/scheduling/v1alpha1/podgroup_webhook.go

```diff
@@ -0,0 +1,337 @@
+	err = a.decoder.Decode(req, rset)
+	if err == nil {
+		err = a.EnsureGroupReplicaSet(rset)
+		if err != nil {
+			logger.Error(err, "Issue adding PodGroup to ReplicaSet")
+			return admission.Errored(http.StatusBadRequest, err)
+		}
+		marshalledSet, err := json.Marshal(rset)
+		if err != nil {
+			logger.Error(err, "Marshalling StatefulSet error")
+			return admission.Errored(http.StatusInternalServerError, err)
+		}
+		logger.Info("Admission StatefulSet success")
+		return admission.PatchResponseFromRaw(req.Object.Raw, marshalledSet)
+	}
+
+	// We should not get down here
+	return admission.Allowed("Object not known, this webhook does not validate beyond those.")
+
+}
+
+// Default is the expected entrypoint for a webhook...
+// I don't remember if this is even called...
+func (a *fluenceWatcher) Default(ctx context.Context, obj runtime.Object) error {
+
+	switch obj.(type) {
+	case *batchv1.Job:
+		job := obj.(*batchv1.Job)
+		return a.EnsureGroupOnJob(job)
+
+	case *corev1.Pod:
+		pod := obj.(*corev1.Pod)
+		return a.EnsureGroup(pod)
+
+	case *appsv1.StatefulSet:
+		set := obj.(*appsv1.StatefulSet)
+		return a.EnsureGroupStatefulSet(set)
+
+	case *appsv1.Deployment:
+		d := obj.(*appsv1.Deployment)
+		return a.EnsureGroupDeployment(d)
+
+	case *appsv1.ReplicaSet:
+		set := obj.(*appsv1.ReplicaSet)
+		return a.EnsureGroupReplicaSet(set)
+
+	default:
+		// no match
+	}
+	return nil
+}
+
+// EnsureGroup adds pod group label and size if not present
+// This ensures that every pod passing through is part of a group.
+// Note that we need to do similar for Job.
+// A pod without a job wrapper, and without metadata is a group
+// of size 1.
+func (a *fluenceWatcher) EnsureGroup(pod *corev1.Pod) error {
+
+	// Add labels if we don't have anything. Everything is a group!
+	if pod.Labels == nil {
+		pod.Labels = map[string]string{}
+	}
+
+	// Do we have a group name?
+	groupName, ok := pod.Labels[labels.PodGroupLabel]
+
+	// If we don't have a fluence group, create one under fluence namespace
+	if !ok {
+		groupName = fmt.Sprintf("fluence-group-%s", pod.Name)
+		pod.Labels[labels.PodGroupLabel] = groupName
+	}
+
+	// Do we have a group size? This will be parsed as a string, likely
+	groupSize, ok := pod.Labels[labels.PodGroupSizeLabel]
+	if !ok {
+		groupSize = "1"
+		pod.Labels[labels.PodGroupSizeLabel] = groupSize
+	}
+	return nil
+}
+
+// getJobLabel takes a label name and default and returns the value
+// We look on both the job and underlying pod spec template
+func getJobLabel(job *batchv1.Job, labelName, defaultLabel string) string {
+
+	value, ok := job.Labels[labelName]
+	if !ok {
+		value, ok = job.Spec.Template.ObjectMeta.Labels[labelName]
+		if !ok {
+			value = defaultLabel
+		}
+	}
+	return value
+}
+
+// EnsureGroupOnJob looks for fluence labels (size and name) on both the job
+// and the pod template. We ultimately put on the pod, the lowest level unit.
+// Since we have the size of the job (paramllism) we can use that for the size
```

Comment:
Ah yes, "paramllism" is the special kind of case where you are eating Parmesan cheese in parallel.

More likely this is muscle memory making a mistake, because "param" is something I'd type a lot. :cheese: